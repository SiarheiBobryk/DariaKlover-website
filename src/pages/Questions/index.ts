export { default } from './Questions';
export { default as questionsMetaData } from './questionsMetaData';
