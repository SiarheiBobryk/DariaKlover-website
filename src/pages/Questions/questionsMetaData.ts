export interface QuestionMetaData {
  title: string;
  heading: string;
  path: string;
}

const questionsMetaData: QuestionMetaData = {
  title: 'Задать вопрос',
  heading: 'Вопросы к Дарье (Бета)',
  path: '/question',
};

export default questionsMetaData;
