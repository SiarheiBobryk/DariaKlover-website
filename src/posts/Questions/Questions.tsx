import * as React from 'react';

import Alert from '@mui/material/Alert';
import AlertTitle from '@mui/material/AlertTitle';

import Paragraph from '../../components/Paragraph';
import Heading from '../../components/Heading';

export interface QuestionsProps {
  heading?: string;
}

function Questions(props: QuestionsProps) {
  const { heading } = props;

  return (
    <>
      <Heading>{heading}</Heading>
      <Paragraph>–ó–¥–µ—Å—å –º–æ–∂–Ω–æ –±—É–¥–µ—Ç –∑–∞–¥–∞—Ç—å –≤–æ–ø—Ä–æ—Å –î–∞—Ä—å–µ</Paragraph>
      <Alert sx={{ my: 1 }} severity="info">
        <AlertTitle>–ö–∞–∫ –∑–∞–¥–∞—Ç—å –≤–æ–ø—Ä–æ—Å</AlertTitle>
        –§–æ—Ä–º–∞, –≥–¥–µ –º–æ–∂–Ω–æ –±—É–¥–µ—Ç –∑–∞–¥–∞—Ç—å –≤–æ–ø—Ä–æ—Å, –∞ —Ç–∞–∫–∂–µ –ø–æ—Å–º–æ—Ç—Ä–µ—Ç—å —á–∞—Å—Ç–æ –∑–∞–¥–∞–≤–∞–µ–º—ã–µ –≤–æ–ø—Ä–æ—Å—ã, —Å–∫–æ—Ä–æ –±—É–¥–µ—Ç –æ–ø—É–±–ª–∏–∫–æ–≤–∞–Ω–∞ –Ω–∞
        —ç—Ç–æ–π —Å—Ç—Ä–∞–Ω–∏—Ü–µ. –°–ø–∞—Å–∏–±–æ –∑–∞ –≤–∞—à–µ —Ç–µ—Ä–ø–µ–Ω–∏–µ üòá
      </Alert>
    </>
  );
}

export default Questions;
