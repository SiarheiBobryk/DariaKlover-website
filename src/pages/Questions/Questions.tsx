import * as React from 'react';
import { Helmet } from 'react-helmet';

import Alert from '@mui/material/Alert';
import AlertTitle from '@mui/material/AlertTitle';

import Heading from '../../components/Heading';
import LinkBlank from '../../components/LinkBlank';
import Paragraph from '../../components/Paragraph';
import questionsMetaData from './questionsMetaData';

function Questions() {
  return (
    <>
      <Helmet>
        <title>{questionsMetaData.title}</title>
      </Helmet>
      <Heading>{questionsMetaData.heading}</Heading>
      <Paragraph>Пожалуйста, перейдите по ссылке ниже. Там будет форма, чтобы задать вопрос Дарье.</Paragraph>
      <Paragraph>
        {/* eslint-disable-next-line max-len */}
        <LinkBlank href="https://docs.google.com/forms/d/e/1FAIpQLSedaIyUajfTsT7ymkNJd8umXnUV-Tc2q6OTmw3Xk8NGB9Hs0A/viewform?usp=sf_link">
          Вопрос Дарье Кловер
        </LinkBlank>
      </Paragraph>
      <Alert severity="warning">
        <AlertTitle>Система тестируется</AlertTitle>
        Обращаем ваше внимание, что страничка, где можно задать вопрос, находится в режиме тестирования и может быть
        нестабильна.
      </Alert>
    </>
  );
}

export default Questions;
