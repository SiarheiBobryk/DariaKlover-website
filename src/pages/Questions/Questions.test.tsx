import * as React from 'react';
import { Helmet, HelmetPropsToState } from 'react-helmet';

import { render, RenderResult } from '@testing-library/react';
import renderer, { ReactTestRendererJSON } from 'react-test-renderer';
import { axe, toHaveNoViolations } from 'jest-axe';

import Questions from './Questions';
import questionsMetaData from './questionsMetaData';

expect.extend(toHaveNoViolations);

describe('<Questions />', () => {
  describe('a11y', () => {
    it('should match a11y standards', async () => {
      const { container }: RenderResult = render(<Questions />);

      const results = await axe(container);
      expect(results).toHaveNoViolations();
    });
  });

  describe('snapshotting', () => {
    it('should render correctly', () => {
      const about: ReactTestRendererJSON | ReactTestRendererJSON[] | null = renderer.create(<Questions />).toJSON();
      expect(about).toMatchSnapshot();
    });
  });

  describe('Elements inspection', () => {
    it('should have the right title HTML element', () => {
      render(<Questions />);

      const { title }: HelmetPropsToState = Helmet.peek();
      expect(title).toBe(questionsMetaData.title);
    });

    it('should have the `h2` HTML element as heading', () => {
      render(<Questions />);

      const h2: HTMLHeadingElement | null = document.querySelector('h2');
      expect(h2).toBeInTheDocument();
      expect(h2).toHaveTextContent(questionsMetaData.heading);
    });
  });
});
