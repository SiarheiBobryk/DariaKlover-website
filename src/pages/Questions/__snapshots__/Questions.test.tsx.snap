// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Questions /> snapshotting should render correctly 1`] = `
Array [
  <h2
    className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
  >
    Вопросы к Дарье (Бета)
  </h2>,
  <p
    className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-d3wcwz-MuiTypography-root"
  >
    Пожалуйста, перейдите по ссылке ниже. Там будет форма, чтобы задать вопрос Дарье.
  </p>,
  <p
    className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-d3wcwz-MuiTypography-root"
  >
    <a
      className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
      href="https://docs.google.com/forms/d/e/1FAIpQLSedaIyUajfTsT7ymkNJd8umXnUV-Tc2q6OTmw3Xk8NGB9Hs0A/viewform?usp=sf_link"
      onBlur={[Function]}
      onFocus={[Function]}
      target="_blank"
    >
      Вопрос Дарье Кловер
       
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-1xzgpfl-MuiSvgIcon-root"
        data-testid="OpenInNewIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
        />
      </svg>
    </a>
  </p>,
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardWarning MuiAlert-standard css-rob5aa-MuiPaper-root-MuiAlert-root"
    role="alert"
  >
    <div
      className="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
        data-testid="ReportProblemOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 5.99L19.53 19H4.47L12 5.99M12 2L1 21h22L12 2zm1 14h-2v2h2v-2zm0-6h-2v4h2v-4z"
        />
      </svg>
    </div>
    <div
      className="MuiAlert-message css-1pxa9xg-MuiAlert-message"
    >
      <div
        className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom MuiAlertTitle-root css-yslrua-MuiTypography-root-MuiAlertTitle-root"
      >
        Система тестируется
      </div>
      Обращаем ваше внимание, что страничка, где можно задать вопрос, находится в режиме тестирования и может быть нестабильна.
    </div>
  </div>,
]
`;
